src/shared/parameter_v2/BUILD

```diff
@@ -0,0 +1,17 @@
+load(":parameter.bzl", "generate_dynamic_parameters")
+
+generate_dynamic_parameters(
+    name = "dynamic_parameters_cpp",
+    enum_targets = [
+        "//software/sensor_fusion:refbox_data",
```

Comment:
I don't understand what this is used for

Reply:
This is an example of how we need to include any targets that define an enum, in order to support it natively in the DynamicParam system.

Eg. If we wanted to autogenerate a `Parameter` that used an enum type, such as `RefboxGameState`, we need to include the header that `RefboxGamestate` is defined in so that we can build the autogenerated parameter code.

Because it's difficult to specify filepaths to other packages in bazel (without doing things like `exporting` files which would add confusion / complexity), we use paths to the targets where the enum is defined. The custom rule is able to extract the headers used to build these targets, giving us the include paths to add to the autogenerated code.

Lmk if that makes sense.

Reply:
If we're reading the enums to create the dynamic param, then I was confused about the naming, because `enum_target` seems to imply that it's an output of this rule. How about `enum_srcs` or `enum_deps`?

Reply:
`enum_deps` is a better idea

Reply:
renamed to `enum_deps`